### PR TITLE
Bookstore dto

### PR DESCRIPTION
**Pull request by azzilian**
BookDto - created as a response DTO
CreateBookRequestDto - as a request DRO
pom - added dependencies & plugins & properties for mapstruck & lombok
BookController - added methods getAll(), getBookById, createBook
Updated service & repository with new method
New exception added - EntityNotFoundException
Bean removed from OnlineBookStoreApplication
application.properties - added jackson.deserialization
BookMapper added - using MapStruct to format toDto and toModel

### CODE REVIEW

File: src/main/java/com/onlinebookstore/onlinebookstore/dto/CreateBookRequestDto.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package com.onlinebookstore.onlinebookstore.dto;
+
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Data
+public class CreateBookRequestDto {
+    private Long id;
```
Comment (Recdt) [inline src/main/java/com/onlinebookstore/onlinebookstore/dto/CreateBookRequestDto.java]: I think this id aren't getted from request, so maybe better to delete it there

Comment (Recdt) [reply src/main/java/com/onlinebookstore/onlinebookstore/dto/CreateBookRequestDto.java]: image
in post only this parameters xd

File: src/main/java/com/onlinebookstore/onlinebookstore/exeption/EntityNotFoundException.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package com.onlinebookstore.onlinebookstore.exeption;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ResponseStatus(HttpStatus.NOT_FOUND)
+public class EntityNotFoundException extends RuntimeException {
```
Comment (Recdt) [inline src/main/java/com/onlinebookstore/onlinebookstore/exeption/EntityNotFoundException.java:3]: Maybe 1 constructor for message + exception ("", e) would be good xd

File: src/main/java/com/onlinebookstore/onlinebookstore/repository/impl/BookRepositoryImpl.java [hunk 3]
```diff
@@ -48,7 +50,15 @@ public List<Book> findAll() {
             Query<Book> query = session.createQuery("FROM Book", Book.class);
             return query.getResultList();
         } catch (Exception e) {
-            throw new DataProcessingException("Can not find any books ", e);
+            throw new EntityNotFoundException("Can not find any books");
+        }
+    }
+
+    @Override
+    public Optional<Book> findBookById(Long id) {
```
Comment (Recdt) [inline src/main/java/com/onlinebookstore/onlinebookstore/repository/impl/BookRepositoryImpl.java]: findBookById more than we need if we are in BookRepository (findById would be enough)

File: src/main/java/com/onlinebookstore/onlinebookstore/repository/impl/BookRepositoryImpl.java [hunk 4]
```diff
@@ -48,7 +50,15 @@ public List<Book> findAll() {
...
+            throw new EntityNotFoundException("Can not find any books");
+        }
+    }
+
+    @Override
+    public Optional<Book> findBookById(Long id) {
+        try (Session session = sessionFactory.openSession()) {
+            Book book = session.find(Book.class, id);
+            return Optional.ofNullable(book);
         }
```
Comment (Recdt) [inline src/main/java/com/onlinebookstore/onlinebookstore/repository/impl/BookRepositoryImpl.java:64]: Maybe catch block or why we need try(if you don't take into account that it's a try with resources)

File: src/main/java/com/onlinebookstore/onlinebookstore/service/BookService.java [hunk 5]
```diff
@@ -1,10 +1,13 @@
...
 import java.util.List;
 
 public interface BookService {
-    Book save(Book book);
+    BookDto save(CreateBookRequestDto requestDto);
 
-    List<Book> findAll();
+    List<BookDto> findAll();
+
+    BookDto findBookById(Long id);
```
Comment (Recdt) [inline src/main/java/com/onlinebookstore/onlinebookstore/service/BookService.java]: BookService findBookById

**Review by Recdt — CHANGES_REQUESTED** [review]
GJ! Rejected because maybe you want to change thg from comments before review.

File: src/main/java/com/onlinebookstore/onlinebookstore/repository/impl/BookRepositoryImpl.java (hunk 4 above)
Comment (azzilian) [reply src/main/java/com/onlinebookstore/onlinebookstore/repository/impl/BookRepositoryImpl.java:64]: It supoused to allow Spring to atomaticaly catch those exceptions and proceed them with @ ResponseStatus

File: src/main/java/com/onlinebookstore/onlinebookstore/controller/BookController.java [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping(value = "/books")
```
Comment (andrii-hoienko) [inline src/main/java/com/onlinebookstore/onlinebookstore/controller/BookController.java]:
```suggestion
@RequestMapping(value = "api/books")

```

File: src/main/java/com/onlinebookstore/onlinebookstore/repository/impl/BookRepositoryImpl.java (hunk 4 above)
Comment (andrii-hoienko) [reply src/main/java/com/onlinebookstore/onlinebookstore/repository/impl/BookRepositoryImpl.java:64]: Agree, please add catch block

**Review by andrii-hoienko — CHANGES_REQUESTED** [review]
Gj! Left minor comments

File: src/main/java/com/onlinebookstore/onlinebookstore/repository/impl/BookRepositoryImpl.java [hunk 7]
```diff
@@ -56,6 +59,8 @@ public Optional<Book> findById(Long id) {
         try (Session session = sessionFactory.openSession()) {
             Book book = session.find(Book.class, id);
             return Optional.ofNullable(book);
+        } catch (Exception e) {
+            throw new EntityNotFoundException("Can't find book by id " + id);
```
Comment (karma-o) [inline src/main/java/com/onlinebookstore/onlinebookstore/repository/impl/BookRepositoryImpl.java]: we can have all sorts of exceptions caught here, and not every one of them would mean that an entity with such an id does not exist. We should throw a simple RuntimeEx or something like `DataProcessingException` like we had before

**Review by karma-o — APPROVED** [review]
Left an important comment, please take a look at it before merging

**Pull request merged**